pennylane/devices/default_qubit_torch.py

```diff
@@ -165,11 +169,37 @@ def __init__(self, wires, *, shots=None, analytic=None):
 
     def execute(self, circuit, **kwargs):
         ops_and_obs = circuit.operations + circuit.observables
-        if any(data.is_cuda for op in ops_and_obs for data in op.data if hasattr(data, "is_cuda")):
-            self._torch_device = "cuda"
+
+        some_params_cuda = any(
+            data.is_cuda for op in ops_and_obs for data in op.data if hasattr(data, "is_cuda")
+        )
+        if not self._torch_device_specified:
+            if some_params_cuda:
+
+                self._torch_device = "cuda"
+            else:
+
+                # need to reset in case last execution moved to cuda
+                self._torch_device = "cpu"
+
+            # If we've changed the device, place the state on the correct
+            # device
+            if self._state.device != self._torch_device:
+                self._state = self._state.to(self._torch_device)
         else:
-            # need to reset in case last execution moved to cuda
-            self._torch_device = "cpu"
+            cuda_params_but_not_cuda_specified = some_params_cuda and self._torch_device != "cuda"
+            not_cuda_params_but_cuda_specified = (
+                not some_params_cuda and self._torch_device == "cuda"
+            )
+            if cuda_params_but_not_cuda_specified or not_cuda_params_but_cuda_specified:
+
+                warnings.warn(
```

Comment:
Great, I think it is now clear for the user that the device in the PennyLane device is the one that will be used 💯